### ISSUE TITLE
full_width = TRUE breaks column alignment with large number of columns

### ISSUE THREAD

**Issue by sharlagelfand** [haozhu233/kableExtra#375]
Maybe related to #372 -- it looks like the column alignment in `kable` breaks when there are "too many" columns and `full_width` in `kable_styling()` is set to `TRUE`. It seems to work for some columns, but not all! I've annotated the output below 😋 

The alignment still works when `full_width = FALSE`, and it also works when `full_width = TRUE` *and* the table is in landscape. 

````
---
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(kableExtra)
library(dplyr)
```

```{r wide-mtcars}
wide_mtcars <- mtcars %>%
  bind_cols(mtcars[,1:5]) %>%
  mutate_all(as.character) 
```

```{r no-full-width}
wide_mtcars %>%
  kable(align = rep("r", ncol(wide_mtcars)),
        caption = "No full width") %>%
  kable_styling(full_width = FALSE)
```

```{r full-width}
wide_mtcars %>%
  kable(align = rep("r", ncol(wide_mtcars)),
        caption = "Full width") %>%
  kable_styling(full_width = TRUE)
```

```{r full-width-landscape}
wide_mtcars %>%
  kable(align = rep("r", ncol(wide_mtcars)),
        caption = "Full width and landscape") %>%
  kable_styling(full_width = TRUE) %>%
  landscape()
```
````

![alignment fine](https://user-images.githubusercontent.com/15895337/56522443-67660000-6514-11e9-9cf0-8d489abfe9d8.PNG)

![align broken](https://user-images.githubusercontent.com/15895337/56522459-6b921d80-6514-11e9-8cc7-d29b74b8aa4b.PNG)

![align landscape](https://user-images.githubusercontent.com/15895337/56522482-70ef6800-6514-11e9-8700-ee2f7c51c0f5.PNG)

**Comment by haozhu233** [comment]
Hi, @sharlagelfand, thank you so much for reporting this bug. I looked into that. It seems like this is a behavior of the latex package `tabu`, which we used here to get the full width effect. The latex code for table 2 & 3 are basically the same. We are using `>{\raggedleft}X` to left align because it's more universal. I also tried to switch the alignment to tabu's `X[m,r]` notation (middle valign, right align) but I got the same result. 🤔

Unfortunately `tabu` this package itself is no longer under active maintenance since 2011 (but there are some attempts for small fixes from the latex core https://github.com/tabu-fixed/tabu, https://tex.stackexchange.com/questions/430973/tabu-m-column-vertical-alignment-not-working-after-package-updates). If I can isolate this problem more precisely, they might be willing to help us out. Let's see.
